### PR TITLE
Fix compile time Absinthe.Plug race condition

### PR DESCRIPTION
**Pull request by chrismcg**
The `if Code.ensure_loaded?(Absinthe.Plug)` line removed by this PR is evaluated at compile time, not run time. This means if `absinthe_plug` is compiled later than this library the `traced_pipeline`
method will not be present.

It could work locally due to `absinthe_plug` being added and compiled before this module. However if there is a different compilation order in e.g. a CI build the function is not added causing an `UndefinedFunctionError` failure.

### CODE REVIEW

File: lib/opencensus/absinthe/plug.ex
Comment (garthk) [inline lib/opencensus/absinthe/plug.ex:20]: I copied the technique here from [`/apollo_tracing/pipeline.ex`](https://github.com/sikanhe/apollo-tracing-elixir/blob/58a91ea/lib/apollo_tracing/pipeline.ex#L8). (Not appealing to authority; just citing refs, and maybe they have some other mitigation against the race condition you mention.)

Is it appropriate to `rescue` … `e in UndefinedFunctionError` and `raise` a custom exception to let people know exactly what's going on? If not, maybe a note in the README they'll find while searching for `UndefinedFunctionError` while troubleshooting.

File: mix.exs
Comment (garthk) [inline mix.exs:50]: My tests are that comprehensive? I only just got to writing the Phoenix integration test in `opencensus_honeycomb`. :/

**Review by garthk — APPROVED** [review]
Looking good; feel free to rebase merge if it'll let you.

File: lib/opencensus/absinthe/plug.ex
Comment (chrismcg) [reply lib/opencensus/absinthe/plug.ex:20]: I'm happy to hold off merging till it's clear the best way to proceed here.

In the original version of the code here https://github.com/sikanhe/apollo-tracing-elixir/blame/9fa7965f33c49913782b620d491b9100a56a607c/lib/apollo_tracing/pipeline.ex it was a runtime check. 

This https://github.com/sikanhe/apollo-tracing-elixir/pull/10#discussion_r157346314 is where it was changed so that library possibly has the bug since then. Though because it's _a_pollo and not _z_pollo maybe that's why it hasn't been much of a problem...

File: mix.exs
Comment (chrismcg) [reply mix.exs:50]: I will check again but pretty sure I had to do this to make the tests pass.

Comment (chrismcg) [reply mix.exs:50]: It was actually to remove a warning:

```
warning: function Absinthe.Plug.default_pipeline/2 is undefined (module Absinthe.Plug is not available)
  lib/opencensus/absinthe/plug.ex:27
```

**Comment by chrismcg** [comment]
I don't have a merge button here

File: lib/opencensus/absinthe/plug.ex
Comment (hauleth) [reply lib/opencensus/absinthe/plug.ex:20]: Instead of removing this line you can add `absinthe_plug` as optional runtime dependency to force compilation of it before building `opencensus_absinthe`.

**Review by hauleth — CHANGES_REQUESTED** [review]

File: lib/opencensus/absinthe/plug.ex
Comment (chrismcg) [reply lib/opencensus/absinthe/plug.ex:20]: :+1: I will do this when I next get chance, likely not for a week or two.

Do you know the reasoning behind not defining the function at all? Is it to be able to warn at compile time if it's used instead of erroring at runtime?

**Comment by lytedev** [comment]
@chrismcg any update on this? mind if I implement @hauleth's changes?

File: mix.exs
Comment (seancribbs) [reply mix.exs:50]: You might want `optional: true` instead of `only: [:dev, :test]`.

**Comment by chrismcg** [comment]
@lytedev please do! I have just started a new gig and don’t have time right now so be great if you can move it forward.

File: lib/opencensus/absinthe/plug.ex
Comment (garthk) [reply lib/opencensus/absinthe/plug.ex:20]: I think they were trying to make it impossible to get a runtime error calling `Absinthe.Plug.default_pipeline/1`.

**Pull request closed without merging**
